Doppler.ContactPolicies.Api.Test/UpdateContactPoliciesSettingsTest.cs

```diff
@@ -0,0 +1,181 @@
+using AutoFixture;
+using Doppler.ContactPolicies.Business.Logic.DTO;
+using Doppler.ContactPolicies.Business.Logic.Extensions;
+using Doppler.ContactPolicies.Business.Logic.Services;
+using Doppler.ContactPolicies.Data.Access.Entities;
+using Doppler.ContactPolicies.Data.Access.Repositories.ContactPoliciesSettings;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Doppler.ContactPolicies.Api.Test
+{
+    public class UpdateContactPoliciesSettingsTest : IClassFixture<WebApplicationFactory<Startup>>
+    {
+        const string TOKEN_ACCOUNT_123_TEST1_AT_TEST_DOT_COM_EXPIRE_20330518 =
+            "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOjEyMywidW5pcXVlX25hbWUiOiJ0ZXN0MUB0ZXN0LmNvbSIsInJvbGUiOiJVU0VSIiwiZXhwIjoyMDAwMDAwMDAwfQ.E3RHjKx9p0a-64RN2YPtlEMysGM45QBO9eATLBhtP4tUQNZnkraUr56hAWA-FuGmhiuMptnKNk_dU3VnbyL6SbHrMWUbquxWjyoqsd7stFs1K_nW6XIzsTjh8Bg6hB5hmsSV-M5_hPS24JwJaCdMQeWrh6cIEp2Sjft7I1V4HQrgzrkMh15sDFAw3i1_ZZasQsDYKyYbO9Jp7lx42ognPrz_KuvPzLjEXvBBNTFsVXUE-ur5adLNMvt-uXzcJ1rcwhjHWItUf5YvgRQbbBnd9f-LsJIhfkDgCJcvZmGDZrtlCKaU1UjHv5c3faZED-cjL59MbibofhPjv87MK8hhdg";
+
+        private const string CONTACT_POLICIES_SETTINGS =
+            "{\"accountName\":\"test1@test.com\",\"active\":true,\"emailsAmountByInterval\":5,\"intervalInDays\":10,\"excludedSubscribersLists\":[]}";
+
+        private readonly WebApplicationFactory<Startup> _factory;
+
+        public UpdateContactPoliciesSettingsTest(WebApplicationFactory<Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        [Theory]
+        [InlineData("test1@test.com", CONTACT_POLICIES_SETTINGS, TOKEN_ACCOUNT_123_TEST1_AT_TEST_DOT_COM_EXPIRE_20330518,
+            HttpStatusCode.NotFound)]
+        public async Task
+            UpdateContactPoliciesSettings_Should_ReturnNotFound_When_UserWithSameAccountNameIsNotFound(
+                string accountName, string contactPoliciesRequestBody, string token, HttpStatusCode expectedStatusCode)
+        {
+            // Arrange
+            int? expectedNotFoundedIdUser = null;
+
+            var contactPoliciesMock = new Mock<IContactPoliciesService>();
+            contactPoliciesMock.Setup(x => x.GetIdUserByAccountName(accountName)).ReturnsAsync(expectedNotFoundedIdUser);
```

Comment:
JC creo que nos conviene usar un `It.IsAny<string>()` para que, independientemente del `accountName`, verifiquemos que se retorne un `NotFound`

```suggestion
        [Theory]
        [InlineData(CONTACT_POLICIES_SETTINGS, TOKEN_ACCOUNT_123_TEST1_AT_TEST_DOT_COM_EXPIRE_20330518,
            HttpStatusCode.NotFound)]
        public async Task
            UpdateContactPoliciesSettings_Should_ReturnNotFound_When_UserWithSameAccountNameIsNotFound(
                string contactPoliciesRequestBody, string token, HttpStatusCode expectedStatusCode)
        {
            // Arrange
            int? expectedNotFoundedIdUser = null;
            string accountName = It.IsAny<string>();
           
            var contactPoliciesMock = new Mock<IContactPoliciesService>();
            contactPoliciesMock.Setup(x => x.GetIdUserByAccountName(accountName)).ReturnsAsync(expectedNotFoundedIdUser);
```

Reply:
usaremos el AutoFixture en su lugar, me parece buena idea